test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -2193,6 +2193,174 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
+            $warnings | Should -BeNullOrEmpty
+            $null = Invoke-WebRequest $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+            $null = Invoke-WebRequest $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+        }
+
+        It 'SSL settings must be set explicitly to break persistence' {
+            $uri = Get-WebListenerUrl
+            $null = Invoke-WebRequest $uri -SessionVariable Session -SslProtocol Tls12 -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+            $null = Invoke-WebRequest $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+            $null = Invoke-WebRequest $uri -WebSession $Session -SslProtocol Default -WarningVariable warnings
+            $warnings | Should -Be $connWarning
+            $null = Invoke-WebRequest $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+            $null = Invoke-WebRequest $uri -WebSession $Session -SslProtocol Default -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+        }
+
+        It 'proxy can turn on and off between tests' {
+            $uri = Get-WebListenerUrl
+            $null = Invoke-WebRequest $uri -SessionVariable Session -WarningVariable warnings
+            $proxy = 'http://127.0.0.1:8080'
+
+            $null = Invoke-WebRequest $uri -WebSession $Session -Proxy $proxy -WarningVariable warnings
+            $warnings | Should -Be $connWarning
+
+            $null = Invoke-WebRequest $uri -WebSession $Session -Proxy $proxy -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+
+            # No proxy at all - use previous setting
+            $null = Invoke-WebRequest $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+
+            # NoProxy toggles proxy off - disconnect
+            $null = Invoke-WebRequest $uri -WebSession $Session -NoProxy -WarningVariable warnings
+            $warnings | Should -Be $connWarning
+
+            # No setting at all - retains NoProxy setting
+            $null = Invoke-WebRequest $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+
+            $null = Invoke-WebRequest $uri -WebSession $Session -Proxy $proxy -WarningVariable warnings
+            $warnings | Should -Be $connWarning
+
+            $null = Invoke-WebRequest $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+
+            $null = Invoke-WebRequest $uri -WebSession $Session -Proxy 'http://localhost:8080' -WarningVariable warnings
+            $warnings | Should -Be $connWarning
+        }
+    }
 }
 
 Describe "Invoke-RestMethod tests" -Tags "Feature", "RequireAdminOnWindows" {
```

Comment:
We have to duplicate the tests for Invoke-RestMethod cmdlet.

Reply:
Let's get it right for IWR then I'll look at duplicating.